index.js

```diff
@@ -6,6 +6,7 @@ var walk = require('jade-walk');
 module.exports = link;
 function link(ast) {
   assert(ast.type === 'Block', 'The top level element should always be a block');
+  if (!ast.nodes.length) return ast;
```

Comment:
This still needs to set `declaredBlocks` to the empty object.  Otherwise extending an empty template will throw an error.